ethpm_cli/_utils/solc.py

```diff
@@ -43,12 +47,53 @@ def generate_solc_input(contracts_dir: Path) -> None:
         contracts_dir.parent,
         SOLC_INPUT,
     )
-    cli_logger.info(
-        f"Use `solc --standard-json --allow-paths /[ABS_PATH_TO]/{contracts_dir} "
-        f"< /[REL_PATH_TO]/{SOLC_INPUT} > /[REL_PATH_TO]/{contracts_dir.parent / SOLC_OUTPUT}` "
-        "to generate the Solidity compiler output. "
-        "Requires that you have the correct Solidity compiler version installed."
+
+
+def compile_contracts(project_dir: Path) -> None:
```

Comment:
I *might* recommend splitting this into two functions.

```python
def find_contracts(root_dir: Path) -> Tuple[Path, ...]:
    ...

def find_solidity_compiler() -> Path:
    ...

def compile_contracts(solc_path: Path: root_dir: Path, source_paths: Collection[Path]) -> None:
    ...
```

This decouples finding the sources from the compilation which are two somewhat distinct phases of the process.  This code is probably fine as is, but I wanted to point this out as a general architectural pattern to look out for.

One small benefit is that you can then decouple the `INFO` level logging from the low level functions.

```python
def cli_compile_comtracts(...):
    paths_to_compile = find_contracts(...)
    logger.info("Compiling contracts: ...")
    if 'ETHPM_SOLC_PATH' in os.environ:
        ...
    else:
        try:
            solc_path = find_solc_path()
            logger.debug("Using solc compiler executable: %s', solc_compiler)
        FileNotFoundError:
            logger.info("No solidity compiler found...")
    compile_contracts(solc_path, ...)
    logger.info("Finished compiling, ...")
```